pkg/controllers/provisioning/launcher.go

```diff
@@ -45,6 +45,16 @@ type Launcher struct {
 	CloudProvider cloudprovider.CloudProvider
 }
 
+// Thread safe channel to pop off packed pod slices
+func queueFor(pods [][]*v1.Pod) <-chan []*v1.Pod {
```

Comment:
What does this code do? Is it trying to avoid the latency of a for loop and start creating immediately? It's not actually clear how this works at all -- isn't the queue closed immediately?

Reply:
The channel is loaded and then closed since all items have been placed on it. After it is closed consumers can still dequeue pod batches, the channel just can't receive anymore. So this is just a thread-safe queue so that the CP can consume batches concurrently.